libcfnet/classic.c

```diff
@@ -58,15 +58,12 @@ static bool LastRecvTimedOut(void)
  *         than #toget bytes only if the peer closed the connection or timeout
  *         or other unrecoverable error occurred.
  */
-int RecvSocketStream(int sd, char buffer[CF_BUFSIZE], int toget)
+SocketRecvStatus RecvSocketStream(int sd, char buffer[CF_BUFSIZE], int toget)
 {
     int already, got;
 
-    if (toget > CF_BUFSIZE - 1 || toget <= 0)
-    {
-        Log(LOG_LEVEL_ERR, "Bad software request to receive %d bytes", toget);
-        return -1;
-    }
+    assert(toget > 0);
+    assert(toget < CF_BUFSIZE - 1);
```

Comment:
Sure you don't wanna leave this in? In release mode you will skip right over this check and go unprotected into the next block.